### ISSUE TITLE
flask example working with wsgi

### ISSUE THREAD

**Issue by molavy** [nickoala/telepot#198]
i try run flask example with wsgi and nginx
so i changed sys.args to manual set variable.
telegram never response , 
i see 200 in nginx response and when i open url in browser i see 200
but i never get telegram response user

**Comment by DanielSerdyukov** [comment]
Hi! I confirm. message_loop() with source queue not working with wsgi.

**Comment by nickoala** [comment]
Please confirm, are you guys using *uWSGI*?

I don't have much experience with wsgi, uwsgi, or nginx. but a web search seems to link nginx with uwsgi (with a 'u' in front). That's why I want to confirm. Thanks.

**Comment by naweiss** [comment]
Same here, DelegatorBot doesn't seems to work.

**Comment by funnyflowerpot** [comment]
Hi @nickoala, I can confirm this problem for uWSGI. I tried [this example](https://github.com/nickoala/telepot/blob/master/examples/webhook/flask_counter.py), but the `request.data` that I `webhook.feed()` does not seem to get forwarded to `MessageCounter`.
 
I'm using [PythonAnywhere](https://www.pythonanywhere.com/), which features a [tutorial for telepot](https://blog.pythonanywhere.com/148/) without `DelegatorBot`. Somebody on their forums tried [the same thing](https://www.pythonanywhere.com/forums/topic/11669/) and apparently a staff member claimed (currently last post in that thread), that this approach won't work due to `telepot`'s need for an own event-handling loop.

PythonAnywhere [uses](https://blog.pythonanywhere.com/36/) uWSGI and nginx. The log-files for my webapp on PythonAnywhere start with these lines:

```
2018-01-28 14:22:21 *** Starting uWSGI 2.0.14 (64bit) on [Sun Jan 28 14:22:18 2018] ***
2018-01-28 14:22:21 compiled with version: 5.4.0 20160609 on 22 November 2017 21:24:09
2018-01-28 14:22:21 os: Linux-4.4.0-109-generic #132-Ubuntu SMP Tue Jan 9 19:52:39 UTC 2018
```

Do you think `DelegatorBot` can be made working with uWSGI? Any help appreciated, thanks.

**Comment by funnyflowerpot** [comment]
Update: In the log files I have just seen a line saying:
```
*** Python threads support is disabled. You can enable it with --enable-threads ***
```
As `telepot` uses threads, this might be the cause of the problem.

**Comment by nickoala** [comment]
Yes, `DelegatorBot` in telepot's traditional version uses threads. Any environments that disallow Python threads would not be able to run `DelegatorBot`.

I personally don't use PythonAnywhere, so can't suggest much here. Either get it to allow threads there, or use telepot's async version, which is based on asyncio (not threads).

**Comment by funnyflowerpot** [comment]
Thanks for the quick reply. Threads are available on PythonAnywhere to non-free accounts only, unfortunately. `asyncio` raised some hopes, but according to [a post from 2015](https://www.pythonanywhere.com/forums/topic/2027/) the used uWSGI implementation doesn't support this either.
As PythonAnywhere is suggesting to use `telepot`, I ask bring this issue to the folks there.
